vg/vgpdf/vgpdf.go

```diff
@@ -25,123 +32,230 @@ const DPI = 72
 // Canvas implements the vg.Canvas interface,
 // drawing to a PDF.
 type Canvas struct {
-	doc         *pdf.Document
-	w, h        vg.Length
-	page        *pdf.Canvas
-	lineVisible bool
+	doc  *pdf.Fpdf
+	w, h vg.Length
+
+	dpi       int
+	numImages int
+	stack     []context
+	fonts     map[vg.Font]struct{}
+}
+
+type context struct {
+	fill  color.Color // fill color
```

Comment:
The comments don't need to exist anymore.

Reply:
Done.